.github/CONTRIBUTING.md

```diff
@@ -1,32 +1,122 @@
+thorough issue description. :brain:
+
+Some points to consider when opening an issue:
+- **Fully complete the provided issue template.** The bug report and feature request templates
+    specify all the information we need to quickly and efficiently address your issue. Be clear,
+    concise, and descriptive.
+    Provide as much information as you can, including steps to reproduce, stack traces, etc.
+- **Use [GitHub-flavored Markdown](https://help.github.com/en/github/writing-on-github/basic-writing-and-formatting-syntax).**
+    Especially put code blocks and console outputs in backticks (```). This improves readability.
+- **Do not open a duplicate feature request.** Search for existing feature requests first. If you
+    find your feature (or one very similar) previously requested, comment on that issue.
+
+## :lock: Reporting Security Issues
+
+**Do not file a public issue for security vulnerabilities.** Please contact the 
+maintainers directly.
+see the [Security Policy](https://github.com/checkstyle/SECURITY.md) for more information.
+
+## :technologist: Submitting Pull Requests
+
+ - **Read our pull request rules.** See [PR Rules](https://github.com/checkstyle/checkstyle/wiki/PR-rules).
+ - **Comment on the issue.** If you are working on an existing issue, please comment on the issue
+    to let others know that you are working on it ("I am on it.").
+ - **Read the Github docs.** Visit GitHub's [Pull Request Guide](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests)
+    for information on how to submit a pull request.
+ - **Follow the template.** Please follow the [CheckStyle Pull Request Template](https://github.com/checkstyle/checkstyle/blob/master/.github/PULL_REQUEST_TEMPLATE.md)
+    that is provided in the pull request description when submitting a pull request.
+ - **Run maven build locally.** `mvn clean verify` should pass on your local before
+   submitting a pull request.
+
+## :white_check_mark: Code Review
+
+All submissions, including submissions by project members, require review. We use GitHub pull
+requests for this purpose. Consult the [GitHub Help](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-request-reviews)
+for more information on pull request reviews.
+
+Here are some general guidelines to follow when submitting a pull request:
+ - **Reply to comments.** If a reviewer asks for changes, make the changes and reply to each
+    (and every)
+    comment to let the reviewer know that you have addressed their concerns.
+ - **Keep the PR small.** If you are working on a large feature, consider breaking it up into
+    smaller PRs that can be reviewed and merged independently. This makes it easier for
+    reviewers to understand the changes and for maintainers to merge the PR.
+ - **Be patient.** Reviewing PRs takes time. If a reviewer hasn't responded in a few days,
```

Comment:
`few days` => `week`. Otherwise, I think we should be specific on the number of days to wait so we aren't constantly pestered.

Reply:
yep, done